lib/cache/cache.go

```diff
@@ -107,6 +116,7 @@ func ForRemoteProxy(cfg Config) Config {
 //
 // ForOldRemoteProxy sets up watch configuration for older remote proxies.
 func ForOldRemoteProxy(cfg Config) Config {
+	cfg.target = "remote-proxy-old"
```

Comment:
You didn't introduce this bug, but while testing this PR I found it. Do you mind removing the line `{Kind: services.KindKubeService}` from `ForOldRemoteProxy`?